### DISCUSSION TITLE
Comparison to GUN?

### DISCUSSION THREAD

**Discussion by rabble** [earthstar-project/earthstar#230] [Q&A]
Hey @cinnamon-bun can you add a comparison to how earthstar differs from the [gun protocol](https://github.com/amark/gun)?

**Comment by cinnamon-bun** [comment]
@rabble I haven't been able to make sense of Gun's documentation.  I'm happy to answer questions about Earthstar though.

Here's a new overview of Earthstar I just wrote: https://github.com/cinnamon-bun/earthstar/blob/master/docs/vocabulary.md

**Comment by xuhcc** [comment]
Here's a relatively short overview of GunDB: https://jaredforsyth.com/posts/local-first-database-gun-js/
From what I can tell, the major differences are:
- Gun uses advanced confict resolution techniques whereas Earthstar resolves conflicts by timestamp.
- Gun tries to find peers automatically (https://gun.eco/docs/Service-Discovery) whereas Earthstar relies solely on pubs.

@cinnamon-bun Is that correct?

**Comment by cinnamon-bun** [comment]
@xuhcc Thanks for finding that.  Your summary seems about right.

Earthstar...
* Designed for many small disconnected workspaces, not a single giant interconnected space like Gun or SSB is.
* Resolves conflicts per document using simple timestamps
* Relies on pub servers.  We'd like to support hyperswarm for direct p2p connections but haven't had the time yet.  Also interested in webrtc or websocket star topologies.  Any way peers can connect they can sync, we just haven't built them all yet. 
* Has a data model of "key-value store holding small JSON documents" similar to CouchDb or Mongo
* Designed to allow partial sync of the data, but not quite implemented yet
* Documents are mutable and deletable
* Has document-level access control.  It would be easy to add pub-level access control also, and we plan for cryptographically enforced workspace-level access control
* Data is signed but there's no built-in encryption yet, but it would be easy to add since user identities are keypairs
* Storage support: sqlite in node, localStorage in browser, indexedDb "soon"
* Can work 100% offline if the host app can (e.g. as a PWA webapp, or a native app)
* Does not yet do anything smart to work well in multiple tabs (which share the same localStorage data).
